nav2_motion_primitives/src/back_up.cpp

```diff
@@ -79,15 +75,15 @@ nav2_tasks::TaskStatus BackUp::controlledBackup()
 
   double diff_x = initial_pose_->pose.pose.position.x - current_odom_pose->pose.pose.position.x;
   double diff_y = initial_pose_->pose.pose.position.y - current_odom_pose->pose.pose.position.y;
-  double distance = sqrt(diff_x * diff_x + diff_y * diff_y);
+  double distance_2 = diff_x * diff_x + diff_y * diff_y;
 
-  if (distance >= abs(command_x_)) {
+  if (distance_2 >= abs(command_x_2_)) {
     cmd_vel.linear.x = 0;
     robot_->sendVelocity(cmd_vel);
     return TaskStatus::SUCCEEDED;
   }
   // TODO(mhpanah): cmd_vel value should be passed as a parameter
-  command_x_ < 0 ? cmd_vel.linear.x = -0.025 : cmd_vel.linear.x = 0.025;
+  command_x_2_ < 0 ? cmd_vel.linear.x = -0.025 : cmd_vel.linear.x = 0.025;
```

Comment:
I would just leave it to be `command_x_` so that we don't loose information on direction.  But I would like to dynamically decrease the linear velocity as we get closer to goal.  Something similar to what you have in your spin method:
```cpp
 double vel = distance_*distance_
 vel = std::min(std::max(vel, min_linear_vel_), max_linear_vel_); 
 command_x_ < 0 ? cmd_vel.linear.x = -vel : cmd_vel.linear.x = vel;
```

Reply:
So there's a little history why I didnt do that. I have set up a number of custom recoveries on some side projects and when I tried to scale the drive backwards recovery, it was really awkward looking. I much preferred the straight acceleration and stopping. The move backwards motion is moving _so_ slow that trying to scale the speed is basically meaningless anyhow. This is 2.5 cm/s.

Reply:
I will change back to command_x_. I missed the signed portion.